components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -861,14 +862,22 @@ private String resolveRegistryURI(String key) {
             if (resourcePath.startsWith(URL_SEPARATOR)) {
                 resourcePath = resourcePath.substring(1);
             }
-
             resolvedPath = registryRoot + resourcePath;
-        }
 
+            //Test whether registry key has any illegel access
+            File resolvedPathFile = new File(resolvedPath);
+            File registryRootFile = new File(registryRoot);
+            try {
+                if (!resolvedPathFile.getCanonicalPath().startsWith(registryRootFile.getCanonicalPath())) {
+                    handleException("The registry key defined is illegal");
```

Comment:
Should we log the resolved path? and the registry root path?

Reply:
In this case, we do not need to. This method will be called several times for various resources. Maybe we can add a debug log. WDYT?

Reply:
+1 to print registry root path but I'm not sure we can print the illegal path in the logs.

Reply:
Logged the registry key